extensions/oidc/runtime/src/main/java/io/quarkus/oidc/AuthenticationContext.java

```diff
@@ -0,0 +1,34 @@
+package io.quarkus.oidc;
+
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.TYPE;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Annotation which can be used to enforce required authentication level for the endpoint classes and methods.
+ * <a href="https://datatracker.ietf.org/doc/rfc9470/">OAuth 2.0 Step Up Authentication Challenge Protocol</a>
+ * is initiated when the access token does not have sufficient authentication strength.
+ */
+@Target({ TYPE, METHOD })
+@Retention(RetentionPolicy.RUNTIME)
+public @interface AuthenticationContext {
+
+    /**
+     * By default, max age is not checked.
+     */
+    long NO_MAX_AGE = -1;
+
+    /**
+     * Required `acr` (Authentication Context Class Reference) claim values.
+     */
+    String[] value();
+
+    /**
+     * If set, the required maximum token age is determined by adding the value of this attribute to the 'auth_time' claim.
+     * The maximum age is set in seconds.
```

Comment:
Can we make it a `Duration` or if not possible - use  `minutes` ? Seconds is too small, tokens can be valid for days

Reply:
Annotation method cannot return `Duration`, it could return string parsed as Duration later like we do in config. Minutes is too much, if users want to specify seconds, they will have to use floats and it is more difficult to specify seconds. E.g. if I want to say 25 seconds, I'd have to set `0.416` of minutes :shrug: .

I can make it `string` (AKA `Duration`) if you want, but I think we can't go with minutes considering specs says that `max_age` is specified in seconds.

Reply:
We have a token age treated as Duration. Even though it is represented as seconds at the spec level, users should be able to use more realistic units without having to convert it to seconds, I don't believe anyone uses tokens valid for 45 seconds etc. 
Indeed, treating it as a Duration format would be easier IMHO

Reply:
Done.